### ISSUE TITLE
Coerce-structure and records gives unpleasant results

### ISSUE THREAD

**Issue by jellelicht** [exoscale/coax#20]
`coerce-structure` looks like a pretty interesting function for playing with spec, but if the result of coercing some value is a record, you get pretty ugly results.

Ex in cljs: 
```clojure
(defrecord Foo [left right])
(defn foo? [o]
  (instance? Foo o))

(defn create-foo-from-input [input]
  (apply ->Foo (str/split input #",")))

(s/def ::foo foo?)
(sc/def ::foo create-foo-from-input)

(sc/coerce-structure {::foo "a,b"})
;; => #:my-ns{:foo ([:right "b"] [:left "a"])}

;; I would expect either:
;; {:my-ns/foo {:left "a", :right "b"}}
;; or rather:
;; {:my-ns/foo #my-ns.Foo{:left "a", :right "b"}}
```
It seems this happens because an instance of `Foo` satisfies `map?`, but calling `empty` on an instance of `Foo` evaluates to `nil`.

**Comment by mpenet** [comment]
Thanks for the report. That could/should indeed be fixed.

**Comment by mpenet** [comment]
I have fixed the roundtrip of records from input -> output. 

Your example was slightly wrong, as coercers take 2 args (second being the opts passed to the coerce call).

```clj
(defrecord Foo [left right])
(defn foo? [o]
  (instance? Foo o))

(defn create-foo-from-input [input _opts]
  (apply ->Foo (str/split input #",")))

(s/def ::foo foo?)

(exoscale.coax/def ::foo create-foo-from-input)

(exoscale.coax/coerce-structure {::foo "a,b"})

#:exoscale.coax{:foo #exoscale.coax.Foo{:left "a", :right "b"}}
```

**Comment by mpenet** [comment]
now available as `coax-1.0.0-alpha16`.

Thanks
